data-processing/entities/sentences/extractor.py

```diff
@@ -45,16 +45,19 @@ def parse(self, tex_path: str, tex: str) -> Iterator[Sentence]:
 
         if last_segment is not None:
             plaintext_to_tex_offset_map[len(plaintext)] = last_segment.tex_end
-
         # Segment the plaintext. Return offsets for each setence relative to the TeX input
         segmenter = pysbd.Segmenter(language="en", clean=False, char_span=True)
+        # Record the previous sentence end to account for extractor bug
+        pre_sentence_end = -1
         for i, sentence in enumerate(segmenter.segment(plaintext)):
-
-            start = plaintext_to_tex_offset_map[sentence.start]
-            end = plaintext_to_tex_offset_map[sentence.end]
+            length = plaintext_to_tex_offset_map[sentence.end] - plaintext_to_tex_offset_map[sentence.start]
```

Comment:
@zkirby this `length` computation doesn't seem quite right.

As an example, consider the two sentences `hi \m. 123) hi.` Because plaintext strips out macros (like `\m`), the characters, their indexes in the TeX, and their indexes in the plain text will be:

```
character   TeX index    Plaintext index
=========   =========    ===============
h           0            0
i           1            1
            2            2
\           3            -
m           4            -
.           5            3
            6            4
1           7            5
2           8            6
3           9            7
)           10           8
            11           9
h           12           10
i           13           11
.           14           12
```

Say we're looking for the `length` of the second sentence, but `pysbd` incorrectly sets `sentence.start, sentence.end` to (0, 8) (wrong start offset, correct length). This would get mapped back to TeX positions 0-10 (follow the map from the "Plaintext index" to the "TeX index" column above). Based on this, the length of `123) hi` would be estimated to be "10 - 0" = 10, which is two more than it's supposed to be. Do I have the right interpretation of how length is getting calculated?

Reply:
Hmm I think I might be a tad confused, would this be solved by just using the sentence.start and sentence.end instead? Or am I misunderstanding

Reply:
Or I am confused by the current tests, which appear to include the macros in the length calculation? Take `test_extract_sentences` for example: 
```
extractor = SentenceExtractor()
 sentences = list(
    extractor.parse(
            "main.tex",
            "This is the first \\macro[arg]{sentence}. This is the second sentence.",
        )
    )
sentence1 = sentences[0]
assert sentence1.start == 0
assert sentence1.end == 40
assert sentences[0].text == "This is the first argsentence."
```
Which shows that the macros are taken into account for the end calculation

Reply:
Yup, that's right! The sentence positions returned by the SentenceExtractor should include the macros. The only thing is, if `pysbd` returns the wrong `sentence.start` and `setnence.end` position (i.e. returns a `start` of 0, as in the example above), if there are macros in the TeX, `plaintext_to_tex_offset_map[sentence.end] - plaintext_to_tex_offset_map[sentence.start]` might return the wrong sentence length.

Reply:
I am realizing now - unless I am misunderstanding - we can't actually do a full tex-string-search right? Bc we don't actually split on the sentences with tex in them so we don't actually have that sentence and in some cases, it will be impossible to find that sentence